Do not fall back to ignoring empty directories for skip when empty - Take 2 (#21625)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

- An input file collection which can't be resolved.

> ==== Gradle does not ignore empty directories for file-trees with SkipWhenEmpty

Reviewer: ```suggestion
==== Gradle does not ignore empty directories for file-trees with `@SkipWhenEmpty`
```

> Gradle detected if an input file collection annotated with `@SkipWhenEmpty` consists only of file trees and then ignored directories automatically.

Reviewer: ```suggestion
Previously Gradle used to detect if an input file collection annotated with `@SkipWhenEmpty` consisted only of file trees and then ignored directories automatically.
```

> To ignore directories in Gradle 8.0 and later, the input property needs to be annotated with `@IgnoreEmptyDirectories`.

Reviewer: ```suggestion
To ignore directories in Gradle 8.0 and later, the input property needs to be explicitly annotated with `@IgnoreEmptyDirectories`.
```
